File: python/other/File.py
filename = 'temp.txt'

# r   只读，文件不存在报错
# w   只写，文件不存在创建
# a   追加，文件不存在追加
# x   只写，文件存在报错
# t   文本模式
# b   二进制模式
# +   读写
file = open(filename, 'a')
file.close()

# write()         只能写字符串
# writelines()    可以写字符串以及字符序列
# => 0
# => 1
# => 2
# => 3
# => 4
with open(filename, 'w') as file:
    if file.writable():
        for i in range(5):
            file.write(str.format('{:d}\n', i))
            print(i)

# read()      读取指定数量的字符
# readline()  读取一行，包括\n
# readlines() 读取所有，返回序列，包括\n
# => 0
# => 1
# => 2
# => 3
# => 4
with open(filename, 'r') as file:
    if file.readable():
        for line in file:
            print(line, end='')

# 截断文件
with open(filename, 'w') as file:
    if file.writable():
        file.truncate(0)

# seek() offset表示偏移量，whence表示起点(0：头，1：当前，2：尾)
# => 3
with open(filename, 'r') as file:
    if file.readable():
        file.seek(0, 2)
        print(file.tell())

File: go/other/file.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"log"
	"os"
	"time"
)

func writeFile(filename string) {
	if file, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY, 0755); err != nil {
		log.Fatal(err)
	} else {
		defer file.Close()

		buf := bufio.NewWriter(file)

		for i := 0; i < 10; i++ {
			buf.WriteString(fmt.Sprintf("%d %s\n", i, time.Now().Format("2006-01-02 15:04:05.000")))
			time.Sleep(time.Millisecond * 10)
		}
		buf.Flush()
	}
}

func readFile(filename string) {
	if file, err := os.OpenFile(filename, os.O_RDONLY, 0755); err != nil {
		log.Fatal(err)
	} else {
		defer file.Close()

		buf := bufio.NewReader(file)

		for {
			data, _, eof := buf.ReadLine()
			if eof == io.EOF {
				break
			}
			fmt.Println(string(data))
		}
	}
}

func main() {
	const filename = "temp.txt"
	writeFile(filename)
	readFile(filename)
}
